### ISSUE TITLE
"Unsolvable" messages worse than "Unsolved"

### ISSUE THREAD

**Issue by robdockins** [GaloisInc/cryptol#768]
While working with some internals related to the new typeclasses, I noticed that when typeclass solvers return `Unsolvable`, the message is considerably less helpful than when the solver just returns `Unsolved`.

For example, when the `Integral` class solver returns `Unsolved` in the default case, I got this message:
```
[error] at Primitive/Asymmetric/Signature/ECDSA/ECDSA.cry:84:3--90:45:
  Failed to validate user-specified signature.
    in the definition of 'Primitive::Asymmetric::Signature::ECDSA::ECDSA::ec_decompress', at Primitive/Asymmetric/Signature/ECDSA/ECDSA.cry:84:3--84:16,
    we need to show that
      for any type Primitive::Asymmetric::Signature::ECDSA::ECDSA::q,
                   Primitive::Asymmetric::Signature::ECDSA::ECDSA::p, a, b
      assuming
        • Primitive::Asymmetric::Signature::ECDSA::ECDSA::Constraints Primitive::Asymmetric::Signature::ECDSA::ECDSA::q
        • Primitive::Asymmetric::Signature::ECDSA::ECDSA::Constraints Primitive::Asymmetric::Signature::ECDSA::ECDSA::p
        • fin a
        • a >= 2 + width Primitive::Asymmetric::Signature::ECDSA::ECDSA::p
        • fin b
        • Primitive::Asymmetric::Signature::ECDSA::ECDSA::p >= 1 + b
      the following constraints hold:
        • Integral (Z Primitive::Asymmetric::Signature::ECDSA::ECDSA::p)
            arising from
            use of expression (Cryptol::%)
            at Primitive/Asymmetric/Signature/ECDSA/ECDSA.cry:90:19--90:28
```

This is pretty verbose, but does precisely pinpoint the `%` operator (and it's location) as the source of the error.

However, If I return `Unsolvable` with a message explaining why, I simply get:

```
[error] at Primitive/Asymmetric/Signature/ECDSA/ECDSA.cry:84:21--90:45:
  Type 'Z Primitive::Asymmetric::Signature::ECDSA::ECDSA::p' is not an intergral type.
```

Which does a great deal less to tell me how to solve the problem.  The `Unsolvable` case should probably print essentially the same message as the `Unsolved` case, but additionally print the message provided.

**Comment by yav** [comment]
Another weird thing there is that for some reason some of the type variables are qualified???

**Comment by robdockins** [comment]
I think that's because they are module parameters, but the whole thing is pretty difficult to read

**Comment by yav** [comment]
Should be fixed via f6213aff3382
